### PR TITLE
Wd scala test update

### PR DESCRIPTION
**Pull request by wjdingman1**
What has been done:
- Create a new test wrapper wdl -> `TestExomeGermlineSingleSample`
- Added a new Tester type to the scala framework `CloudWorkflowTester`
- Added command line parameters for that new tester type

This PR allows the scala testing framework to take in a pipeline now as an argument rather than a command.

A more thorough write up can be found [here](https://docs.google.com/document/d/1ydfZOBArLqh4hHzrgQdc_M73lT1G-5NKRB3LM9BqEzs/edit#) in the post mortem section.

### CODE REVIEW

File: tasks/broad/CopyFilesFromCloudToCloud.wdl [hunk 1]
```diff
@@ -22,7 +22,7 @@ task CopyFilesFromCloudToCloud {
     String vault_token_path
     String google_account_vault_path
     Float? contamination
-    String base_file_name
+    String base_file_name = "base_file"
```
Comment (nikellepetrillo) [inline tasks/broad/CopyFilesFromCloudToCloud.wdl:25]: did you mean to hard code "base_file" here? maybe there is some kind of input id to pass here

Comment (wjdingman1) [reply tasks/broad/CopyFilesFromCloudToCloud.wdl:25]: I did actualy, some pipelines will produce outputs that are non-file i.e. for WGS it produces a contamination score which is a float. When this is done you need to copy that float to a file and name it with `base_file_name`...i added a fall back name (which will be unused) for pipelines that don't produce nonfile output

**Review by nikellepetrillo — APPROVED** [review]

File: pipelines/broad/arrays/imputation/Imputation.changelog.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
+# 1.0.8
+2022-04-04 (Date of Last Commit)
+
+* Update task CopyFilesFromCloudToCloud in Utilities.wdl, this update has no effect on this pipeline
 # 1.0.7
-2022-03-08 (Date of Last Commit)
+2022-03-08(Date of Last Commit)
```
Comment (gbggrant) [inline pipelines/broad/arrays/imputation/Imputation.changelog.md:6]: I like the space.

File: tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/config/CloudWorkflowConfig.scala [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  PipelineTestType,
+  WorkflowTestCategory
+}
+
+case class CloudWorkflowConfig(
+    category: WorkflowTestCategory = WorkflowTestCategory.Plumbing,
+    env: CromwellEnvironment = CromwellEnvironment.Dev,
+    pipeline: PipelineTestType = PipelineTestType.ExomeGermlineSingleSample,
+    truthBranch: String = "develop",
+    papiVersion: PapiVersion = PAPIv2,
```
Comment (gbggrant) [inline tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/config/CloudWorkflowConfig.scala:16]: I *think* we can get rid of PapiVersion now.  everything should default to PAPIv2 (and you really shouldn't need to set it)

File: tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/tester/CloudWorkflowTester.scala [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+      workflowName + ".results_path" -> resultsPath.asJson,
+      workflowName + ".update_truth" -> updateTruth.asJson,
+      workflowName + ".vault_token_path" -> vaultTokenPath.asJson,
+      workflowName + ".google_account_vault_path" -> googleAccountVaultPath.asJson
+    )
+
+    /**
+      * If we have nested inputs in our test inputs we need to push them down a level
+      * e.g.
+      * ExomeGermUrllineSingleSample.AggregatedBamQC.CollectReadgroupBamQualityMetrics.collect_gc_bias_metrics ->
```
Comment (gbggrant) [inline tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/tester/CloudWorkflowTester.scala:149]: ExomeGermUrlline??

File: verification/test-wdls/TestExomeGermlineSingleSample.wdl [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                            select_all([ExomeGermlineSingleSample.cross_check_fingerprints_metrics]),
+                            select_all([ExomeGermlineSingleSample.fingerprint_summary_metrics]),
+                            select_all([ExomeGermlineSingleSample.fingerprint_detail_metrics]),
+                            select_all([ExomeGermlineSingleSample.output_bqsr_reports]),
+                            select_all([ExomeGermlineSingleSample.output_bam]),
+                            select_all([ExomeGermlineSingleSample.output_bam_index]),
+  ])
+
+  # Collect all of the pipeline metrics into a single Array[String]
+  Array[String] pipeline_metrics = flatten([
```
Comment (gbggrant) [inline verification/test-wdls/TestExomeGermlineSingleSample.wdl:98]: Are all of pipeline_metrics in pipeline_outputs? If so, could you collect the former first and then add them to pipeline_outputs (where pipeline_outputs initially contains only non-metrics).

**Review by gbggrant — APPROVED** [review]
Looks really good - only minor comments.

File: tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/config/CloudWorkflowConfig.scala (hunk 3 above)
Comment (wjdingman1) [reply tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/config/CloudWorkflowConfig.scala:16]: We don't have a command line flag to set it, we are just assigning it a default value in the config here. Probably just not going to mess with it...

**Pull request merged**